src/Shared/LayoutRenderers/AspNetRequestDurationLayoutRenderer.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using NLog.Config;
+using NLog.LayoutRenderers;
+
+namespace NLog.Web.LayoutRenderers
+{
+    /// <summary>
+    /// Renders the current duration of the active HttpRequest
+    /// </summary>
+    [LayoutRenderer("aspnet-request-duration")]
+    [ThreadSafe]
+    public class AspNetRequestDurationLayoutRenderer : AspNetLayoutRendererBase
+    {
+        private static string[] DurationMsFormat = null;
+#if !ASP_NET_CORE
+        private string _formatString;
+#endif
+
+        /// <summary>
+        /// When no format specified, then just total milliseconds
+        /// </summary>
+        public string Format { get; set; }
+
+        /// <summary>
+        /// Gets or sets the culture used for rendering. 
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+        public System.Globalization.CultureInfo Culture { get; set; } = System.Globalization.CultureInfo.InvariantCulture;
+
+        /// <inheritdoc/>
+        protected override void InitializeLayoutRenderer()
+        {
+            if (DurationMsFormat == null && string.IsNullOrEmpty(Format) && ReferenceEquals(Culture, System.Globalization.CultureInfo.InvariantCulture))
+            {
+                System.Threading.Interlocked.CompareExchange(ref DurationMsFormat, System.Linq.Enumerable.ToArray(System.Linq.Enumerable.Select(System.Linq.Enumerable.Range(0, 1000), i => i.ToString())), null);
```

Comment:
what are we trying to do in `System.Linq.Enumerable.ToArray(System.Linq.Enumerable.Select(System.Linq.Enumerable.Range(0, 1000), i => i.ToString())`? Maybe create a method for it as "documentation"